Sources/AsyncAlgorithms/AsyncThrowingChannel.swift

```diff
@@ -158,53 +171,77 @@ public final class AsyncThrowingChannel<Element: Sendable, Failure: Error>: Asyn
              state.emission = .awaiting(nexts)
            }
           return nil
+        case .terminated(let termination):
+          potentialTermination = termination
```

Comment:
at this point the terminal state is consumed with the failure. So we wouldn't expect to hold onto that error indefinitely; instead it likely should follow the rules of AsyncSequence iteration and return nil for subsequent calls. To that end it probably should set the state.emission to `.terminated(.finished)`

Reply:
good catch! I've adapted both the code and the unit test to reflect that.